cpp/src/arrow/table.h

```diff
@@ -113,7 +114,7 @@ class ARROW_EXPORT RecordBatch {
       const std::vector<std::shared_ptr<Array>>& columns);
 
   RecordBatch(const std::shared_ptr<Schema>& schema, int64_t num_rows,
```

Comment:
The move-based constructor was only being used internally AFAIK, but I have restored it